src/modals/ModalDonate/DonateRecurrent.tsx

```diff
@@ -0,0 +1,384 @@
+    Input,
+    OptionItem,
+    colors
+} from '@impact-market/ui';
+import { Text } from '../../theme/components';
+import { getCookie } from 'cookies-next';
+import { handleKnownErrors } from '../../helpers/handleKnownErrors';
+import { handleSignature } from '../../helpers/handleSignature';
+import { toast } from '../../components/Toaster/Toaster';
+import { useDonationMiner, useSignatures } from '@impact-market/utils';
+import { useTranslation } from '../../components/TranslationProvider/TranslationProvider';
+import React, { useEffect, useState } from 'react';
+import RichText from '../../lib/Prismic/components/RichText';
+import config from '../../../config';
+import styled from 'styled-components';
+import { dateHelpers } from 'src/helpers/dateHelpers';
+
+const OptionsStyled = styled(Box)`
+    display: grid;
+    grid-template-columns: repeat(2, 1fr);
+    gap: 1rem;
+
+    a {
+        margin-top: 0 !important;
+    }
+`;
+
+export const EmailStyled = styled(Box)`
+    display: flex;
+    flex-direction: column-reverse;
+
+    p {
+        font-weight: 500;
+        color: ${colors.g700};
+        padding: 0 0 0.375rem 0;
+    }
+`;
+
+const CancelButtonStyled = styled(Button)`
+    background-color: ${colors.e50};
+    color: ${colors.e700};
+    border: none;
+
+    &:hover {
+        background-color: ${colors.e700} !important;
+        color: ${colors.e50} !important;
+        border: none !important;
+    }
+`;
+
+type SelectedOption = {
+    label: string;
+    value: number;
+    months: number;
+};
+
+export const DonateRecurrent = ({
+    recurrentDonation,
+    refresh,
+    setRefresh,
+    setDonationType
+}: any) => {
+    const { t } = useTranslation();
+
+    const { signMessage } = useSignatures();
+
+    const { approve } = useDonationMiner();
+    const [approveIsLoading, setApproveIsLoading] = useState(false);
+    const [depositIsLoading, setDepositIsLoading] = useState(false);
+    const [depositButtonDisabled, setDepositButtonDisabled] = useState(true);
+    const [email, setEmail] = useState('');
+    const [newEmail, setNewEmail] = useState('');
+
+    const getEmail = async () => {
+        try {
+            const res = await fetch(`${config.baseApiUrl}/users`, {
+                headers: {
+                    Accept: 'application/json',
+                    Authorization: `Bearer ${getCookie('AUTH_TOKEN')}`,
+                    'Content-Type': 'application/json'
+                },
+                method: 'GET'
+            });
+
+            const response = await res.json();
+
+            setEmail(response?.data?.email);
+        } catch (error) {
+            console.log('User Data Error: ', error);
+
+            return false;
+        }
+    };
+
+    useEffect(() => {
+        getEmail();
+    }, []);
+
+    const handleApprove = async () => {
+        setApproveIsLoading(true);
+        setDepositButtonDisabled(true);
+
+        try {
+            const response = await approve(selected?.value);
```

Comment:
Don't forget to multiply the approved amount by 10, so it will work for at least 10 months
```suggestion
            const response = await approve(selected?.value * 10);
```